File: lesson-08/js/MyStackRun.js
class MyStack {
    constructor(size) {
        this.storage = new Array(size);
        this.index = 0;
    }

    push(val) {
        this.storage[this.index++] = val;
    }

    pop() {
        return this.storage[--this.index];
    }

    peek() {
        return this.storage[this.index - 1];
    }

    toString() {
        return `[${this.storage.slice(0, this.index).join(", ")}]`;
    }
}

const main = () => {

    const ms = new MyStack(10);
    console.log(ms.toString()); // []
    ms.push(10);
    ms.push(20);
    ms.push(30);

    console.log(ms.toString()); // [10, 20, 30]
    console.log(ms.peek()); // 30
    console.log(ms.peek()); // 30
    console.log(ms.peek()); // 30
    console.log(ms.pop()); // 30
    console.log(ms.toString()); // [10, 20]

    console.log(ms.pop()); // 20
    console.log(ms.toString()); // [10]
}

main();

File: lesson-08/java/MyStackRun.java
import java.util.Arrays;

public class MyStackRun {
    public static void main(String[] args) {
        MyStack ms = new MyStack(10);
        System.out.println(ms);
        ms.push(10);
        ms.push(20);
        ms.push(30);

        System.out.println(ms);
        System.out.println(ms.peek());
        System.out.println(ms.peek());
        System.out.println(ms.peek());
        System.out.println(ms.pop());
        System.out.println(ms);

        System.out.println(ms.pop());
        System.out.println(ms);
    }

}

class MyStack {
    int[] storage;
    int index;

    public MyStack(int size) {
        this.storage = new int[size];
        this.index = 0;
    }

    public void push(int val) {
        storage[index++] = val;
    }

    public int pop() {
        return storage[--index];
    }

    public int peek() {
        return storage[index - 1];
    }

    @Override
    public String toString() {
        return Arrays.toString(Arrays.copyOfRange(storage, 0, index));
    }
}
